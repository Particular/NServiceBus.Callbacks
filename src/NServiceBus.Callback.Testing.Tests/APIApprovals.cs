using NServiceBus.Callbacks.Testing;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(TestableCallbackAwareSession).Assembly);
        Approver.Verify(publicApi);
    }
}