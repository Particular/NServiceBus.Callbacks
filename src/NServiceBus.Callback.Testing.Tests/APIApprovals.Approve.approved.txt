[assembly: ApprovalTests.Reporters.UseReporterAttribute(new System.Type[] {
        typeof(ApprovalTests.Reporters.AllFailingTestsClipboardReporter),
        typeof(ApprovalTests.Reporters.DiffReporter)})]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]
[NUnit.Framework.TestFixtureAttribute()]
public class APIApprovals
{
    public APIApprovals() { }
    [NUnit.Framework.TestAttribute()]
    public void Approve() { }
}
namespace NServiceBus.Callback.Testing.Tests
{
    [NUnit.Framework.TestFixtureAttribute()]
    public class TestablecallbackAwareSessionTests
    {
        public TestablecallbackAwareSessionTests() { }
        [NUnit.Framework.TestAttribute()]
        public void When_request_matcher_does_not_match_should_honor_cancellation() { }
        [NUnit.Framework.TestAttribute()]
        public System.Threading.Tasks.Task When_request_matcher_matches_should_not_honor_cancellation() { }
        [NUnit.Framework.TestAttribute()]
        public System.Threading.Tasks.Task When_request_matcher_matches_should_return_response() { }
        [NUnit.Framework.TestAttribute()]
        public void When_request_matcher_matches_with_wrong_response_type() { }
        [NUnit.Framework.TestAttribute()]
        public void When_request_option_matcher_does_not_match_should_honor_cancellation() { }
        [NUnit.Framework.TestAttribute()]
        public System.Threading.Tasks.Task When_request_option_matcher_matches_should_not_honor_cancellation() { }
        [NUnit.Framework.TestAttribute()]
        public System.Threading.Tasks.Task When_request_option_matcher_matches_should_return_response() { }
        [NUnit.Framework.TestAttribute()]
        public void When_request_option_matcher_matches_with_wrong_response_type() { }
    }
}
namespace PublicApiGenerator
{
    public class static ApiGenerator
    {
        public static string GeneratePublicApi(System.Reflection.Assembly assemby, System.Type[] includeTypes = null, bool shouldIncludeAssemblyAttributes = True) { }
    }
}